tests/sentry/api/test_event_search.py

```diff
@@ -1078,15 +1078,25 @@ def test_project_name(self):
 
         params = {"project_id": [p1.id, p2.id]}
         filter = get_filter("project.name:{}".format(p1.slug), params)
-        filter.conditions == [["project_id", "=", p1.id]]
-        filter.filter_keys == {"project_id": [p1.id, p2.id]}
-        filter.project_ids == [p1.id, p2.id]
+        assert filter.conditions == [["project_id", "=", p1.id]]
+        assert filter.filter_keys == {"project_id": [p1.id, p2.id]}
+        assert filter.project_ids == [p1.id, p2.id]
 
-        params = {"project_id": []}
+        params = {"project_id": [p1.id, p2.id]}
         filter = get_filter("!project.name:{}".format(p1.slug), params)
-        filter.conditions == [["project_id", "!=", p1.id]]
-        filter.filter_keys == {}
-        filter.project_ids == []
+        assert filter.conditions == [
+            [[["isNull", ["project_id"]], "=", 1], ["project_id", "!=", p1.id]]
+        ]
+        assert filter.filter_keys == {"project_id": [p1.id, p2.id]}
+        assert filter.project_ids == [p1.id, p2.id]
+
+        with pytest.raises(InvalidSearchQuery) as err:
+            params = {"project_id": []}
+            filter = get_filter("project.name:{}".format(p1.slug), params)
```

Comment:
```suggestion
            get_filter("project.name:{}".format(p1.slug), params)
```
Variable was unused.